impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java

```diff
@@ -26,7 +26,7 @@
   // that need to be accommodated.
   public static final boolean IS_RESTRICTED_APPENGINE =
       System.getProperty("com.google.appengine.runtime.environment") != null
-          && "1.7".equals(System.getProperty("java.specification.version"));
+          && "1.8".equals(System.getProperty("java.specification.version"));
```

Comment:
I think for Java 8 we don't need to do this. In the Java 8 appengine they removed this requirement. Can you please check that?

Reply:
It looks like AppEngine still does not support renaming of threads.  thread.setName() isn't allowed.

Reply:
ok if that is the case do this:

```
  @Override
  public Thread newThread(Runnable r) {
    Thread thread = threadFactory.newThread(r);
    try {
      thread.setName(name);
      thread.setDaemon(true);
    } catch (SecurityException e) {
      // OK if we can't set the name or daemon in this environment.
    }
    return thread;
  }
```

This is also how guava implements this in `MoreExecutors.newThread()` which unfortunately is not public.

This way we remove the IS_RESTRICTED_APPENGINE and we are ok. What do you think?

Reply:
Yes, we can do that.